flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveSchedulerTest.java

```diff
@@ -1002,6 +1007,72 @@ public void testComputeVertexParallelismStoreForExecutionInDefaultMode() {
         }
     }
 
+    @Test
+    public void testCheckpointCleanerIsClosedAfterCheckpointServices() throws Exception {
+        final JobGraph jobGraph = createJobGraph();
+        enableCheckpointing(jobGraph);
+        final CountDownLatch checkpointServicesShutdownBlocked = new CountDownLatch(1);
+        final CountDownLatch cleanerClosed = new CountDownLatch(1);
+        final ScheduledExecutorService executorService =
+                Executors.newSingleThreadScheduledExecutor();
+        final CompletedCheckpointStore completedCheckpointStore =
+                new StandaloneCompletedCheckpointStore(1) {
+
+                    @Override
+                    public void shutdown(JobStatus jobStatus, CheckpointsCleaner checkpointsCleaner)
+                            throws Exception {
+                        checkpointServicesShutdownBlocked.await();
+                        super.shutdown(jobStatus, checkpointsCleaner);
+                    }
+                };
+        final CheckpointIDCounter checkpointIDCounter =
+                new StandaloneCheckpointIDCounter() {
+
+                    @Override
+                    public void shutdown(JobStatus jobStatus) throws Exception {
+                        checkpointServicesShutdownBlocked.await();
+                        super.shutdown(jobStatus);
+                    }
+                };
```

Comment:
You might be able to deduplicate these classes with the test for the `DefaultScheduler`.

Reply:
I've created a static version of the test in `DefaultSchedulerTest` so it can be reused.